### PR TITLE
OSSMDOC-189

### PR DESCRIPTION
**Pull request by neal-timpe**
https://issues.redhat.com/browse/OSSMDOC-189

Ossm release notes. 
Includes https://issues.redhat.com/browse/OSSMDOC-116
https://issues.redhat.com/browse/MAISTRA-1979

Cherrypick this PR to 4.6 and 4.7.
I'll need to create separate PRs for 4.4 and 4.5 that only include 1.1 items.

### CODE REVIEW

File: modules/ossm-rn-fixed-issues.adoc [hunk 1]
```diff
@@ -19,9 +19,21 @@ The following issues been resolved in the current release:
 [id="ossm-rn-fixed-issues-ossm_{context}"]
 == {ProductShortName} fixed issues
 
+* link:https://issues.redhat.com/browse/MAISTRA-2010[MAISTRA-2010] AuthorizationPolicy does not support `request.regex.headers` field. The `validatingwebhook` rejects any AuthorizationPolicy with the field, and even if you disable that, Pilot tries to validate it using the same code, and it does not work. The issue was fixed in 1.1, but not 2.0.
```
Comment (dgn) [inline modules/ossm-rn-fixed-issues.adoc:22]: "The issue was fixed in 1.1, but not 2.0" sounds a bit like we didn't fix the issue after all. maybe just mention that this was a regression? it is fixed in both versions now

File: modules/ossm-rn-new-features.adoc [hunk 2]
```diff
@@ -36,6 +36,10 @@ Result – If changed, describe the current user experience
 |2.0.0
 |===
 
+== New features {ProductName} 2.0.1
+
+This release of {ProductName} adds support for Istio 1.6.5, Jaeger 1.20.0, Kiali 1.24.2, and the 3scale Istio Adapter 2.0 and OpenShift Container Platform 4.6.
```
Comment (dgn) [inline modules/ossm-rn-new-features.adoc:41]: we updated to Istio 1.6.14

File: modules/ossm-vs-istio.adoc [hunk 3]
```diff
@@ -39,6 +39,45 @@ The upstream Istio community installation automatically injects the sidecar into
...
+
+The upstream Istio community installation includes options to perform exact header matches, match wildcards in headers, or check for a header containing a specific prefix or suffix.
+
+{ProductName} extends the ability to match request headers by using a regular expression. Specify a property key of `request.regex.headers` with a regular expression.
+
+.Upstream Istio community matching request headers example
+[source,yaml]
+----
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRoleBinding
```
Comment (dgn) [inline modules/ossm-vs-istio.adoc:55]: ServiceRoleBinding is deprecated in 2.x -- we should show an example that uses AuthorizationPolicy to achieve this. You can find one in the jira issue: https://issues.redhat.com/browse/MAISTRA-2010

**Comment by neal-timpe** [comment]
@dgn Can you take another look at this?

**Review by JStickler — APPROVED** [review]

File: modules/ossm-vs-istio.adoc [hunk 4]
```diff
@@ -39,6 +39,49 @@ The upstream Istio community installation automatically injects the sidecar into
...
+  action: ALLOW
+  rules: 
+    - when: 
+        - key: 'request.regex.headers[username]'
+          values: 
+            - "allowed.*"
+  selector: 
+    matchLabels: 
+      app: httpbin
+----
```
Comment (dgn) [inline modules/ossm-vs-istio.adoc:68]: the AuthorizationPolicy above is the OSSM example

File: modules/ossm-vs-istio.adoc [hunk 5]
```diff
@@ -39,6 +39,49 @@ The upstream Istio community installation automatically injects the sidecar into
...
+kind: ServiceRoleBinding
+metadata:
+  name: httpbin-client-binding
+  namespace: httpbin
+spec:
+  subjects:
+  - user: "cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account"
+    properties:
+      request.regex.headers[<header>]: "<regular expression>"
+----
```
Comment (dgn) [inline modules/ossm-vs-istio.adoc:83]: the ServiceRoleBinding resource is not required anymore. We used to have two resources, ServiceRole and ServiceRoleBinding that were required to do this, but now it's just AuthorizationPolicy, both upstream and in OSSM

File: modules/ossm-vs-istio.adoc (hunk 4 above)
Comment (dgn) [reply modules/ossm-vs-istio.adoc:68]: the upstream one is without the `.regex.` part (see first entry in table here: https://istio.io/latest/docs/reference/config/security/conditions/#supported-conditions)

**Review by dgn — CHANGES_REQUESTED** [review]
still a few things off about the request header matching. description is good, but the examples are wrong

**Comment by neal-timpe** [comment]
@dgn Can you take one more look? Thanks!

File: modules/ossm-rn-fixed-issues.adoc [hunk 6]
```diff
@@ -19,9 +19,21 @@ The following issues been resolved in the current release:
 [id="ossm-rn-fixed-issues-ossm_{context}"]
 == {ProductShortName} fixed issues
 
+* link:https://issues.redhat.com/browse/MAISTRA-2010[MAISTRA-2010] AuthorizationPolicy does not support `request.regex.headers` field. The `validatingwebhook` rejects any AuthorizationPolicy with the field, and even if you disable that, Pilot tries to validate it using the same code, and it does not work.
+
+* link:https://issues.jboss.org/browse/MAISTRA-1979[MAISTRA-1979] _Migration to 2.0_ The conversion webhook drops the following important fields when converting SMCP.status from v2 to v1:
```
Comment (brian-avery) [inline modules/ossm-rn-fixed-issues.adoc:24]: Should this be: 
```suggestion
* link:https://issues.jboss.org/browse/MAISTRA-1979[MAISTRA-1979] _Migration to 2.0_ The conversion webhook drops the following important fields when converting `SMCP.status` from v2 to v1:
```

**Review by brian-avery — APPROVED** [review]

File: modules/ossm-cr-example.adoc [hunk 7]
```diff
@@ -183,7 +183,7 @@ The following table lists the specifications for the `ServiceMeshControlPlane` r
 |Name |Description |Configurable parameters
 
 |addons
-|Addons is used to configure additional features beyond core control plane components, such as visualization, or metric storage.
+|Addon is used to configure additional features beyond core control plane components, such as visualization, or metric storage.
```
Comment (brian-avery) [inline modules/ossm-cr-example.adoc:186]: Shouldn't this be `Addons`? I read this as referring to the field name.... FieldName is used to configure...

**Review by rcernich — APPROVED** [review]
lgtm

**Review by gbaufake — APPROVED** [review]

**Pull request merged**